Source/WebCore/rendering/RenderTheme.cpp

```diff
@@ -462,14 +480,7 @@ bool RenderTheme::paint(const RenderBox& box, ControlStates& controlStates, cons
     ControlPart part = box.style().effectiveAppearance();
     IntRect integralSnappedRect = snappedIntRect(rect);
 
-    // Temporarily move this call above the canPaint check to allow
-    // this to work in the GPU process
-#if ENABLE(ATTACHMENT_ELEMENT)
-    if (part == AttachmentPart || part == BorderlessAttachmentPart)
-        return paintAttachment(box, paintInfo, integralSnappedRect);
-#endif
-
-    if (UNLIKELY(!canPaint(paintInfo, box.settings())))
+    if (!canPaintWithoutPlatformContext(part) && UNLIKELY(!canPaint(paintInfo, box.settings())))
```

Comment:
This is getting hard to follow. What about passing the ControlPart to `canPaint()`?

Reply:
Fixed.